### ISSUE TITLE
selector with square brackets and dot

### ISSUE THREAD

**Issue by dagda1** [airbnb/enzyme#1082]
I have the following selector:

`const nameElement = wrapper.find('[data-selector="previousNames[0].forename"]');`

But I get this error:

> TypeError: Enzyme::Unable to parse selector '[data-selector="previousNames[0]'. Perhaps you forgot to escape a string? Try '[data-selector=""previousNames[0"]' instead.

It should be a valid selector because this works in the browser:

`document.querySelector('[data-selector="previousNames[0].forename"]')`

This works as a workaround:

```
        const nameElement = wrapper
          .find('input[type="text"]')
          .findWhere(
            x => x.props()['data-selector'] === 'previousNames[0].forename'
          );
```

Would it be worth trying to fix this?

**Comment by ljharb** [comment]
If it can be fixed with minimal effort and a regression test, yes! Otherwise it's probably best to leave it as-is until we can add an actual selector engine to enzyme.

Separately, non-string APIs are always better choices than string selector APIs ;-)

**Comment by aweary** [comment]
I have a branch with a working integration of an actual CSS selector parser, and I confirmed that it successfully parses this selector! I'll self-assign so I remember it when I open the PR.

**Comment by ljharb** [comment]
Closed in #1086.
